src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,87 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] buckets;
+    private double threshold;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new Node[INITIAL_CAPACITY];
+        threshold = INITIAL_CAPACITY * LOAD_FACTOR;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = getIndex(key);
+        Node<K, V> node = buckets[index];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        Node<K, V> newNode = new Node<>(key, value);
+        newNode.next = buckets[index];
+        buckets[index] = newNode;
+        size++;
+        threshold = (double) size / buckets.length;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> node = buckets[index];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        if (key == null) {
+            return 0;
+        }
+        return Math.abs(key.hashCode()) % buckets.length;
+    }
+
+    private void resize() {
```

Comment:
My suggested solution and yours are involve resizing and rehashing, but the first block also resets the size and creates a completely new table, while the second block rehashes elements into a new array without resetting the size or creating a new table from scratch.
You as developer could decide which solution is more suitable for you. I just point you to places that can be optimized or be different.

Reply:
Thanks! Makes sense. I will try out your suggestions)